source/fluid_pressure_boundary_conditions/interface.cc

```diff
@@ -0,0 +1,160 @@
+/*
+  Copyright (C) 2015 - 2016 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/global.h>
+#include <aspect/fluid_pressure_boundary_conditions/interface.h>
+
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/std_cxx1x/tuple.h>
+
+#include <list>
+
+
+namespace aspect
+{
+  namespace FluidPressureBoundaryConditions
+  {
+    template <int dim>
+    Interface<dim>::~Interface ()
+    {}
+
+    template <int dim>
+    void
+    Interface<dim>::initialize ()
+    {}
+
+    template <int dim>
+    void
+    Interface<dim>::
+    declare_parameters (dealii::ParameterHandler &/*prm*/)
+    {}
+
+
+    template <int dim>
+    void
+    Interface<dim>::parse_parameters (dealii::ParameterHandler &/*prm*/)
+    {}
+
+
+// -------------------------------- Deal with registering models and automating
+// -------------------------------- their setup and selection at run time
+
+    namespace
+    {
+      std_cxx1x::tuple
+      <void *,
+      void *,
+      internal::Plugins::PluginList<Interface<2> >,
+      internal::Plugins::PluginList<Interface<3> > > registered_plugins;
+    }
+
+
+
+    template <int dim>
+    void
+    register_fluid_pressure_boundary (const std::string &name,
+                                      const std::string &description,
+                                      void (*declare_parameters_function) (ParameterHandler &),
+                                      Interface<dim> *(*factory_function) ())
+    {
+      std_cxx1x::get<dim>(registered_plugins).register_plugin (name,
+                                                               description,
+                                                               declare_parameters_function,
+                                                               factory_function);
+    }
+
+
+    template <int dim>
+    Interface<dim> *
+    create_fluid_pressure_boundary (ParameterHandler &prm)
+    {
+      std::string model_name;
+      prm.enter_subsection ("Fluid Pressure Boundary Condition");
```

Comment:
Now here it gets messy :smile:. We now have another boundary condition that follows the naming convection of the velocity boundary condition (velocity_boundary_condition) instead of the naming convection of the composition boundary condition (boundary_composition), but uses the input parameter structure of the composition, because the velocity is handled in the "Model parameters" subsection of the parameter file (that I never liked, it is too generic) and does not have its own subsection (except all the derived velocity plugins use "Boundary velocity model"). For reference: I initialized a discussion about this a while ago (#179), but never actually did something about it. Maybe we need to speak about this one more time at the hackathon and finally decide: 1. If we want to unify the handling of naming conventions between the initial and boundary conditions. 2. If we can at least agree on naming conventions that all 'future additions' should follow.

For now: All other boundary conditions name their parameter subsection: "Boundary velocity model" or similar. So "Boundary fluid pressure model" would be most in line with the other plugins. Let me know what you think about that change. If you prefer keeping consistency with your filenames instead of the other plugins you should call it "Fluid pressure boundary conditions". My favored naming approach (as discussed in #179) would be "Boundary fluid pressure", but this would be too avantgarde considering the rest of the current structure :smirk: